capstone/src/main/webapp/script.js

```diff
@@ -110,25 +130,211 @@ async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
+  if (typeof(response.attributeScores) === 'undefined') {
     analysisContainer.removeChild(loadingEl);
     analysisContainer.appendChild(createAnyElement('p', 'Perspective API was not able to get scores for detailed analysis'));
     return;
   }
   const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
   const attributes = sortAttributes(response);
 
-  // Color the substring appropriately	
-  const substringEl = createAnyElement('span', substring);
+  // Set up replacements function call and color the substring appropriately	
+  const substringEl = document.createElement('span');
+  const words = getSubstrings(substring, TokenizerEnum['WORD']);
+  for (let i = 0; i < words.length; i++) {
+    const wordEl = createAnyElement('span', words[i]);
+    const bareWord = words[i].replace(/[\s.,\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // Remove any punctuation & whitespace
+    wordEl.onclick = function() { setUpReplacements(substring, lang, bareWord); }
+    wordEl.classList.add('output-word'); 
+    substringEl.appendChild(wordEl);
+  }
   colorSubstring(substringEl, toxicityScore);
      
   // Attach a tooltip (info-box for the substring)
   const tooltipEl = createTooltip(attributes);
   substringEl.appendChild(tooltipEl);
   result.appendChild(substringEl);
 }
+ 
+/** Sets up the replacements section */
+function setUpReplacements(text, lang, substringForReplacements) {
+  // Clear previous results
+  const headerContainer = document.getElementById('perspective-datamuse-header');
+  headerContainer.innerHTML = '';
+  document.getElementById('perspective-datamuse-chart').innerHTML = '';
+  document.getElementById('perspective-datamuse-extras').innerHTML = '';
+
+  // Set up about container 
+  const aboutContainer = document.createElement('div');
+  aboutContainer.setAttribute('style', 'background-color:#FAEBD7');
+  aboutContainer.appendChild(createAnyElement('p', ABOUT_PARAGRAPH));
+  headerContainer.appendChild(aboutContainer);
+
+  getReplacements(text, lang, substringForReplacements);
+  setUpExtras(text, lang, substringForReplacements);
+}
+
+/** Gets the replacements & their score changes for a subtring based on the Datamuse API */
+async function getReplacements(text, lang, substringForReplacements) {
+  // Clear previous results
+  const analysisContainer = document.getElementById('perspective-datamuse-analysis');
+  analysisContainer.innerHTML = '';
+  document.getElementById('perspective-datamuse-chart').innerHTML = '';
+
+  // Set up loading spinner
+  const loadingEl = document.createElement('p');
+  loadingEl.className = 'spinner-border';
+  analysisContainer.appendChild(loadingEl);
+
+  // Get the word type
+  let wordType = getReplacementsWordType();
+  if (!wordType) {
+    wordType = DATAMUSE_ATTRIBUTES['Means like'];
+  }
+
+  // Get the toxicity score of original string 
+  const toxicityOfOriginal = await getOriginalToxicity(text, lang, analysisContainer, loadingEl);
+  if (!toxicityOfOriginal) {
```

Comment:
Do you want to add an error message here, maybe similar to the case below under `if (replacements.length === 0)`?

Reply:
Yup, added

Reply:
It seems like `getOriginalToxicity` does some similar error handling itself. How do they interact? Do both error messages show up if Perspective doesn't return scores?

Reply:
Yes, I hadn't caught this. Both error messages would show up, so I only left the error handling in `getOriginalToxicity`